solana/README.md

```diff
@@ -0,0 +1,515 @@
+
+```sh
+solana airdrop 2
+```
+
+## Deployments
+
+Setup
+
+1. Clone the solana-axelar repo.
+2. Compile the Solana programs.
+
+> [!IMPORTANT]
+> For the initial deployment of Solana programs to any of the clusters (devnet, testnet, and mainnet-beta), the program keypairs are required. The pubkey is the program ID and is hardcoded in the program using the `declare_id` macro. In case a new set of keypairs is required, a new release of the crates needs to happen afterwards (due to the id being hardcoded). Updating the ids can be done within the `solana-axelar` directory by invoking:
+> ```sh
+> cargo xtask update-ids
+> ```
+> The keypair files should be stored securely as they're needed for the initial deployment on other clusters as well.
+
+> [!NOTE]
+> Initial deployment of Solana programs doesn't support offline signing, the process needs to be done online. When deploying, an `upgrade-authority` can be set, which will later be able to perform program upgrades — upgrades support offline signing.
+
+In order to get verifiable builds, we use `solana-verify` tool. For more information on how to use the tool - including when multisig is used (which is expected as upgrade authority for mainnet deployments) - visit the [Solana guide for verifiable builds](https://solana.com/developers/guides/advanced/verified-builds).
+
+Set the `BASE_IMAGE` variable:
+
+```sh
+export BASE_IMAGE="solanafoundation/solana-verifiable-build@sha256:979b09eef544de4502a92e28a724a8498a08e2fe506e8905b642e613760403d3"
+```
+
+```sh
+# Go to the solana directory within the cloned repo
+pushd solana-axelar
+
+# Compile the Solana programs
+solana-verify build --base-image $BASE_IMAGE --library-name axelar_solana_gas_service
+solana-verify build --base-image $BASE_IMAGE --library-name axelar_solana_gateway
+solana-verify build --base-image $BASE_IMAGE --library-name axelar_solana_governance
+solana-verify build --base-image $BASE_IMAGE --library-name axelar_solana_its
+solana-verify build --base-image $BASE_IMAGE --library-name axelar_solana_multicall
+
+# Go back
+popd
+```
+
+3. Declare enviroment variables:
+
+```sh
+GATEWAY_PROGRAM_KEYPAIR_PATH=<path/to/gateway_program_keypair.json>
+GATEWAY_PROGRAM_PATH="solana-axelar/target/deploy/axelar_solana_gateway.so"
+
+GAS_SERVICE_PROGRAM_KEYPAIR_PATH=<path/to/gas_service_program_keypair.json>
+GAS_SERVICE_PROGRAM_PATH="solana-axelar/target/deploy/axelar_solana_gas_service.so"
+
+GOVERNANCE_PROGRAM_KEYPAIR_PATH=<path/to/governance_program_keypair.json>
+GOVERNANCE_PROGRAM_PATH="solana-axelar/target/deploy/axelar_solana_governance.so"
+
+MULTICALL_PROGRAM_KEYPAIR_PATH=<path/to/multicall_program_keypair.json>
+MULTICALL_PROGRAM_PATH="solana-axelar/target/deploy/axelar_solana_multicall.so"
+
+ITS_PROGRAM_KEYPAIR_PATH=<path/to/its_program_keypair.json>
+ITS_PROGRAM_PATH="solana-axelar/target/deploy/axelar_solana_its.so"
+
+UPGRADE_AUTHORITY_PUBKEY=<base58 pubkey to set as upgrade authority>
+```
```

Comment:
What do you think if i move this to an `.env.example` file, which is commited in the repo. Then i could add instructons to just copy `.env.example` -> `.env` (which is ignored) and adapt it to custom paths.

Then user just need to `source .env`

Let me know if that's ok, and i will perform the change in separate branch pointing to this pr :)

Reply:
Yeah, I think it might be helpful. We just need to be careful since there's already a `.env.example` used by the other integrations and for some checks the user will use more than one integration. Perhaps just adding these variables to the existing `.env.example` under a comment `# Solana` or something like that...